src/login/utils.ts

```diff
@@ -1,13 +1,57 @@
-import { UserCredential } from "@firebase/auth";
-import { NavigateFunction } from "react-router";
+import { User, UserCredential } from "firebase/auth";
+import { doc, DocumentData, DocumentReference, DocumentSnapshot, getDoc } from "firebase/firestore";
+import { NavigateFunction } from "react-router-dom";
+import { db } from "../api/Firebase";
 import { actionTypes } from "../context/reducer";
+import { UserInfo } from "../types/user";
+import { isNullOrUndefined } from "../utils/utils";
 
-export function setUser(dispatch: any, result: UserCredential, navigate: NavigateFunction) {
+export const setUser = async (dispatch: any, result: UserCredential, navigate: NavigateFunction) => {
     dispatch({
       type: actionTypes.SET_USER,
       user: result.user
     });
     const userString = JSON.stringify(result.user);
     sessionStorage.setItem(process.env.REACT_APP_USER_CREDENTIALS || '', userString);
+    await fetchUserInfo(result.user, dispatch)
     return navigate("/posters")
```

Comment:
I think the redirect isn't working, I tried loging in and it's not redericting

Reply:
It is, but now it takes a few seconds because of this fetchUserInfo.
IMO all we can do is create a loading.
Tell me what you think.